### PR TITLE
Extract the Fake UserService from the SubjectRepository test

### PR DESCRIPTION
**Pull request by DGuhr**
precursor for using it also in app_test.go to do real roundtrips with the import API. Also extracts the fakeIdp from app_test.

## Describe your changes

- extract FakeUserApi to reuse it elsewhere
- extract FakeUserIdp to make app_test more test-only.
- ignore it when building release binary
- add certDir to `createTestServer(...)` as parameter to make it configurable from different packages. 

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

**Comment by Rajagopalan-Ranganathan** [comment]
ignore it when building release binary - Where are these changes @DGuhr ?

File: testenv/FakeUserServiceApi.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
+//go:build !release
```
Comment (DGuhr) [inline testenv/FakeUserServiceApi.go:1]: @Rajagopalan-Ranganathan This build flag here ignores the file when the build is tagged with tag `release` which we do in `make binary` and elsewhere. Same as in https://github.com/RedHatInsights/authz/blob/main/infrastructure/repository/authzed/LocalSpiceDbContainer.go#L1 

That said, I just added another commit to rename the `testData` dir to `testdata`, as per the conventions this package is also ignored, see `go help packages`, last section:

> ...Directory and file names that begin with "." or "_" are ignored
by the go tool, as are directories named "testdata".

**Review by Rajagopalan-Ranganathan — APPROVED** [review]

**Review by Rajagopalan-Ranganathan — APPROVED** [review]

**Comment by Rajagopalan-Ranganathan** [comment]
LGTM

**Comment by DGuhr** [comment]
Superseeded by #167

**Pull request closed without merging**